pkgs/development/compilers/p4c/default.nix

```diff
@@ -0,0 +1,85 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, boehmgc
+, bison
+, flex
+, protobuf
+, gmp
+, boost
+, python3
+, doxygen
+, graphviz
+, libbpf
+, libllvm
+, enableDocumentation ? true
+, enableBPF ? true
+, enableDPDK ? true
+, enableBMV2 ? true
+, enableGraphBackend ? true
+, enableP4Tests ? true
+, enableGTests ? true
+, enableMultithreading ? false
+}:
+let
+  toCMakeBoolean = v: if v then "ON" else "OFF";
+in
+stdenv.mkDerivation rec {
+  pname = "p4c";
+  version = "v1.2.2.1";
+
+  src = fetchFromGitHub {
+    owner = "p4lang";
+    repo = "p4c";
+    rev = version;
+    sha256 = "sha256-XIZ7Cm5zfr5XA+s0aSG1WwWHCPjAYv/YoBWTRaXi9rQ=";
+    fetchSubmodules = true;
+  };
+
+  cmakeFlags = [
+    "-DENABLE_BMV2=${toCMakeBoolean enableBMV2}"
+    "-DENABLE_EBPF=${toCMakeBoolean enableBPF}"
+    "-DENABLE_UBPF=${toCMakeBoolean enableBPF}"
+    "-DENABLE_DPDK=${toCMakeBoolean enableDPDK}"
+    "-DENABLE_P4C_GRAPHS=${toCMakeBoolean enableGraphBackend}"
+    "-DENABLE_P4TEST=${toCMakeBoolean enableP4Tests}"
+    "-DENABLE_DOCS=${toCMakeBoolean enableDocumentation}"
+    "-DENABLE_GC=ON"
+    "-DENABLE_GTESTS=${toCMakeBoolean enableGTests}"
+    "-DENABLE_PROTOBUF_STATIC=ON"
+    "-DENABLE_MULTITHREAD=${toCMakeBoolean enableMultithreading}"
+    "-DENABLE_GMP=ON"
+  ];
+
+  checkPhase = ''
+    make check
+  '';
+
+  buildInputs = [
```

Comment:
Please properly sort buildInputs. Flex and bison goes to native while at least boehmgc, boots and gmp go not. Also native is listed before buildInputs.